### DISCUSSION TITLE
DXF conversion to SVG yields error for SPLINE entity

### DISCUSSION THREAD

**Discussion by luoyu-123** [mozman/ezdxf#291] [Issues] [answered]
I used the add_spline method when I used ezdxf to draw the spline curve. When I converted the dxf file to the svg image format, an error occurred. What is the reason?
![image](https://user-images.githubusercontent.com/59160360/103401190-e3866700-4b82-11eb-80cc-0b8cbdb8d73d.png)

**Comment by mozman** [comment]
Please keep the headline short. Use the appropriate boards, this is an issue but can not be transferred to "Issues" as far I know. But it is easy to transfer an "Issue" to "Discussions".

And at last to ALL: Provide the damn source code, the exception traceback alone was never enough to find a bug.

**Comment by mozman** [comment]
Thank you for pointing out that there is no audit functionality implemented for the SPLINE entity at all!

**Comment by mozman** [comment]
Added an audit method to the `Spline` entity, which will prevent the exception as shown in your post, if you use the safe loading procedure as shown in the [documentation](https://ezdxf.mozman.at/docs/addons/drawing.html):

```Python
# Safe loading procedure (requires ezdxf v0.14):
try:
    doc, auditor = recover.readfile('your.dxf')
except IOError:
    print(f'Not a DXF file or a generic I/O error.')
    sys.exit(1)
except ezdxf.DXFStructureError:
    print(f'Invalid or corrupted DXF file.')
    sys.exit(2)
```

But this will *not* prevent **you** from creating invalid splines, if you use the `add_spline()` to create a spline and you don't provide the required data like enough fit points or control points or a too small degree for the given point count. 

As simple rule, use `degree + 1` control points or at least `degree` fit points and  don't use both. 

A spline is only defined by the control points (and the knot values), using fit points is just a convenience method to define a spline, but render algorithm have to calculate the required control points (and knot values) from these fit points. Which is by the way not that easy because fit points do not define a single spline instead an infinite set of splines, from which the algorithm choose one spline, which leads to the problem that ezdxf does not render the same splines as AutoCAD or BricsCAD, because the way this CAD applications calculate the end tangents, if they are not stored in the DXF files, is not known.

Using fit points is an easy way to define splines but you don't know exact what you get, beside the fact that the spline passes through your fit points.

**Comment by luoyu-123** [comment, marked as answer]
非常感谢

**Comment by luoyu-123** [comment]
thanks
